kong/plugins/inflate-gzip/handler.lua

```diff
@@ -24,23 +24,21 @@ function plugin:header_filter(conf)
 
   if conf.deflate_response and (not c_encoding or (not string_find(c_encoding, "gzip", 1, true))) then
     kong.log.debug("will deflate response body - clearing headers")
-
+    kong.response.clear_header("Transfer-Encoding")  -- we aren't chunking anymore
     kong.response.set_header("Content-Encoding", "gzip")
-    kong.response.clear_header("Content-Length")  -- chunk it instead
   end
-end
-
 
-function plugin:body_filter(conf)
   -- check that it isn't already gzip'd
   local c_encoding = kong.request.get_header("Content-Encoding")
   if conf.deflate_response and (not c_encoding or (not string_find(c_encoding, "gzip", 1, true))) then
     kong.log.debug("deflating response")
 
-    local response_body = kong.response.get_raw_body()
+    local response_body = kong.service.response.get_raw_body()
 
     if response_body then
-      kong.response.set_raw_body(deflate_gzip(response_body))
+      -- stash the body for filter later
+      kong.ctx.plugin.compressed_body = deflate_gzip(response_body)
+      kong.response.set_header("Content-Length", #kong.ctx.plugin.compressed_body)
```

Comment:
```suggestion
      local ctx_plugin = kong.ctx.plugin
      ctx_plugin.compressed_body = deflate_gzip(response_body)
      kong.response.set_header("Content-Length", #ctx_plugin.compressed_body)
```

fyi, accessing magic tables is expensive, so local scoped variables can be used to cache them. Just make sure to not cache them beyond a single phase handler function!